CTOR-1183 : Refresh hardware-storage-dell-compellent-api (#3829)
centreon/centreon-documentation: pp/integrations/plugin-packs/procedures/hardware-storage-dell-compellent-api.md

id: hardware-storage-dell-compellent-api
title: Dell Compellent API
---
import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

## Connector dependencies

The following monitoring connectors will be installed when you install the **Dell Compellent API** connector through the
**Configuration > Monitoring Connector Manager** menu:
* [Base Pack](./base-generic.md)

## Pack assets

### Templates

The Monitoring Connector **Dell Compellent API** brings a host template:

* **HW-Storage-Dell-Compellent-NRPE-custom**

The connector brings the following service templates (sorted by the host template they are attached to):

<Tabs groupId="sync">
<TabItem value="HW-Storage-Dell-Compellent-NRPE-custom" label="HW-Storage-Dell-Compellent-NRPE-custom">

| Service Alias | Service Template                                    | Service Description |
|:--------------|:----------------------------------------------------|:--------------------|

> | Hba-Usage     | HW-Storage-Dell-Compellent-Hba-Usage-NRPE-custom    | Check hba usages    |

Reviewer: ```suggestion
| Hba-Usage     | HW-Storage-Dell-Compellent-Hba-Usage-NRPE-custom    | Check hba usage    |
```

> | Volume-Usage  | HW-Storage-Dell-Compellent-Volume-Usage-NRPE-custom | Check volume usages |

Reviewer: ```suggestion
| Volume-Usage  | HW-Storage-Dell-Compellent-Volume-Usage-NRPE-custom | Check volume usage |
```

| EXTRAOPTIONS           | Any extra option you may want to add to the command (a --verbose flag for example). All options are listed [here](#available-options). | --verbose         |             |

</TabItem>
</Tabs>

3. [Deploy the configuration](/docs/monitoring/monitoring-servers/deploying-a-configuration). The service appears in the list of services, and on the **Resources Status** page. The command that is sent by the connector is displayed in the details panel of the service: it shows the values of the macros.

## How to check in the CLI that the configuration is OK and what are the main options for?

Once the plugin is installed, log into your Centreon poller's CLI using the
**centreon-engine** user account (`su - centreon-engine`). Test that the connector 
is able to monitor a resource using a command like this one (replace the sample values by yours):

```bash
/usr/lib64/nagios/plugins//check\_centreon\_nrpe -H  -p 5666 -t 50  -c check_centreon_plugins -a 'storage::dell::compellent::local::plugin' 'volume-usage'  '  \

> 	--cem-user="" \
> 	--cem-password="" \
> 	--cem-host="" \
> 	--cem-port="3033" \
> 	--sdk-path-dll="" \

Author: ```suggestion
	--cem-user="XXXX" \
	--cem-password="XXXX" \
	--cem-host="XXXX" \
	--cem-port="3033" \
	--sdk-path-dll="XXXX" \
```

> > Don't use the '!' character in centreon macro configuration!

Reviewer: ```suggestion
> Do not use the '!' character in when configuring Centreon macros.
```

| EXTRAOPTIONS           | Any extra option you may want to add to the command (a --verbose flag for example). All options are listed [here](#available-options). | --verbose         |             |

</TabItem>
</Tabs>

3. [Deploy the configuration](/docs/monitoring/monitoring-servers/deploying-a-configuration). The service appears in the list of services, and on the **Resources Status** page. The command that is sent by the connector is displayed in the details panel of the service: it shows the values of the macros.

## How to check in the CLI that the configuration is OK and what are the main options for?

Once the plugin is installed, log into your Centreon poller's CLI using the
**centreon-engine** user account (`su - centreon-engine`). Test that the connector 
is able to monitor a resource using a command like this one (replace the sample values by yours):

```bash
/usr/lib64/nagios/plugins//check\_centreon\_nrpe -H  -p 5666 -t 50  -c check_centreon_plugins -a 'storage::dell::compellent::local::plugin' 'volume-usage'  '  \
	--cem-user="" \
	--cem-password="" \
	--cem-host="" \

> 	--cem-port="3033" \

Reviewer: user, password, host?
